cgi-bin/DW/Controller/Talk.pm

```diff
@@ -157,7 +157,11 @@ DW::Routing->register_string( '/talkpost_do', \&talkpost_do_handler, app => 1 );
 #     listens for this by name.
 
 sub talkpost_do_handler {
-    my ( $ok, $rv ) = controller( form_auth => 1, anonymous => 1 );
+
+    # This route handles form_auth manually instead of passing form_auth => 1 to
+    # controller() -- we want to fail gently AFTER we have our post args, so we
+    # can regenerate the comment form and not lose your text.
+    my ( $ok, $rv ) = controller( anonymous => 1 );
```

Comment:
We need to explicitly say `form_auth => 0` here to signal that we are doing our own auth handling. There was discussion of changing the default from 0 to 1 at some point.

Reply:
Ah, thank you! Didn't know that.